test/timers/test-timers-nested.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var common = require('../common');
+var assert = require('assert');
+
+// Make sure we test 0ms timers, since they would had always wanted to run on
+// the current tick, and greater than 0ms timers, for scenarios where the
+// outer timer takes longer to complete than the delay of the nested timer.
+// Since the process of recreating this is identical regardless of the timer
+// delay, these scenarios are in one test.
+var scenarios = [0, 100];
+
+scenarios.forEach(function(delay) {
+  var nestedCalled = false;
+
+  setTimeout(function A() {
+    // Create the nested timer with the same delay as the outer timer so that it
+    // gets added to the current list of timers being processed by
+    // listOnTimeout.
+    setTimeout(function B() {
+      nestedCalled = true;
+    }, delay);
+
+    // Busy loop for the same timeout used for the nested timer to ensure that
+    // we are in fact expiring the nested timer.
+    common.busyLoop(delay);
```

Comment:
Example:

The pattern used here is prone to this sort of error. You should schedule your timeout after you do your work.

A similar thing may happen if, say, you tell an http server to start listening before you attach any handlers to it.

If you schedule a timeout, and it's time passes, and it's in the timeout list, it will timeout as soon as possible. Isn't that expected?